src/DynamoCore/Core/Threading/DynamoScheduler.cs

```diff
@@ -74,6 +74,18 @@ public interface IScheduler
         IEnumerable<AsyncTask> Tasks { get; }
```

Comment:
Thanks @ke-yu, the operation is safe IFF the `Task` are immutable (which I think they still are, for now). The call `AsyncTaskExtension.AllComplete` isn't mutating the tasks themselves so I don't want to hold this PR back any longer, thanks for clarifying it anyway :)